_rest_config.php

```diff
@@ -289,6 +289,47 @@ public static function authorization_check($section, $value): void
         }
     }
 
+    // Main function to check scope
+    //  Use cases:
+    //     Only sending $first would be for something like 'openid'
+    //     For using all 3 parameters would be for something like 'user/Organization.write'
+    //       $first = 'user', $second = 'Organization', $third = 'write'
+    //     Note the 3 parameter method will also take into account wildcard scopes.
+    public static function scope_check($first, $second = null, $third = null): void
+    {
+        if (!empty($GLOBALS['oauth_scopes'])) {
+            // Need to ensure has scope
+            if (empty($second)) {
+                // Simply check to see if $first is an allowed scope
+                $scope = $first;
+                if (!in_array($scope, $GLOBALS['oauth_scopes'])) {
+                    http_response_code(401);
+                    exit;
+                }
+            } else {
+                // More complicated 3 degree scope check with wildcard support
+                $scope[] = $first . '/' . $second . '.' . $third;
+                $scope[] = $first . '/*.' . $third;
+                $scope[] = $first . '/' . $second . '.*';
+                $scope[] = $first . '/*.*';
+                $scope[] = '*/' . $second . '.' . $third;
+                $scope[] = '*/*.' . $third;
+                $scope[] = '*/' . $second . '.*';
+                $scope[] = '*/*.*';
+                $passScope = false;
+                foreach ($scope as $checkScope) {
+                    if (in_array($checkScope, $GLOBALS['oauth_scopes'])) {
+                        $passScope = true;
+                    }
+                }
+                if (!$passScope) {
+                    http_response_code(401);
```

Comment:
I wasn't sure if just passing 401 was sufficient or if we needed to pass anything else to be spec compliant but this good for FHIR R4.  @see https://www.hl7.org/fhir/security.html#AccessDenied